vumi/persist/fields.py

```diff
@@ -56,24 +57,52 @@ def _add_index(self, modelobj, value):
             return
         modelobj._riak_object.add_index(self.index_name, str(value))
 
+    def get_riak_data(self, modelobj, default=None, key=None):
+        if key is None:
+            key = self.key
+        return modelobj._riak_object.get_data().get(key, default)
+
+    def set_riak_data(self, modelobj, raw_value, key=None):
+        if key is None:
+            key = self.key
+        old_raw_value = modelobj._riak_object.get_data().get(key)
+        # We set this even if it's "unmodified" because we want explicit `None`
+        # values rather than missing fields.
+        modelobj._riak_object.set_data_field(key, raw_value)
```

Comment:
Yes. The whole object is a single entity in Riak and it gets written unconditionally whenever `save()` is called.

Trying to detect changes (which could be to indexes or metadata rather than just the object itself) is more complexity than we want, especially since writes of unmodified objects are much less common than writes of modified objects. The only reason we're bothering with it here is to avoid loops where computed values trigger each other's update functions despite not actually depending on each other.